### ISSUE TITLE
[Bug] Unable to register / create account

### ISSUE THREAD

**Issue by Ejz9** [AmruthPillai/Reactive-Resume#1438]
### Is there an existing issue for this?

- [X] Yes, I have searched the existing issues

### Product Variant

Self-Hosted

### Current Behavior

I run the default docker-compose listed in Reactive-Resume Git Hub. I am deploying the docker-compose through portainer. Portainer creates a stack with the Postgres-db, reactResume server, and reactResume client. In the logs, the db shows nothing of error. The server shows nothing of error. The client shows nothing of error.

Yet when I attempt to "register" I am greeted with the inability to. The status bar when you click "register" crosses the just below my search bar a couple of times before disappearing. The screen does not change otherwise. There is no hint of what might be wrong. I've re-deployed the stack many times attempting various ideas or potential solutions I thought might work. I have also deployed the stack with adminer so I could see if the db was actually getting written to. When you register an account, it is not getting written to. Yet some point during the setup/startup of everything the server must have some connection to the db cause it creates "users" and "resumes" tables. There are no entries in these tables. I have even tried entering the data to make an account manually. No resolve.

I've seen this issue has persisted for a long time and previous reports were open about the same thing. I also believe the docker guide is a bit lackluster. The docs that exist do guide you somewhat but when you already have a docker-compose. A tool that many individuals already use, why is there not a section for it in the docs, or at least how to set it up? Docker makes deploying something like this incredibly easy, simple, and convenient. But the existing docker-compose just does not work, or there's some error that many other individuals like myself are similarly having. It's discouraging to deal with.

The most common cause? I'm presuming some database-to-server error. They communicate initially it seems but fail to maintain a connection afterward or there is some hiccup. Noting is printed in logs (at least docker logs for each container). You can visibly see when it is not able to connect though as if you changed ex: the db-host env variable to something other than "postgres" the server spits out the issue in its logs.

### Expected Behavior

I would expect that I could create an account on my self-hosted version of this program. Which other issues might be presented but presumably would be minor and easily fixed. I would also expect the logs to state when an error has occurred with anything.

### Steps To Reproduce

1. Grab or copy this docker-compose file: https://github.com/AmruthPillai/Reactive-Resume/blob/main/compose.yml
2. Deploy with whatever tool you manage your docker-compose files with. I used portainer and added it as an app template. From this, I adjusted the external ports for myself and deployed the stack to which no issues were shown by portainer.
3. Visit the site: default is http://your-ip/your-domain:3000 to visit the client landing page. Mine was port 5000 for outside of the container. It still brings you to the same place.
4. Go to the top left of the page where it says register and fill out the information.
5. Click register after filling out the required information. Watch for a status bar at the top (just below the search bar) and expect that it does not proceed with creating the account or displaying anything more.

### What browsers are you seeing the problem on?

Firefox, Chrome, Microsoft Edge

### Anything else?

I'm doing this on OperaGX but it did not work on the other three browsers when I tested. I do not believe it is a browser issue.

**Comment by darsh12** [comment]
I am having the same issues:

```yaml
# .env
PUBLIC_URL=192.168.68.201:3000
PUBLIC_SERVER_URL=192.168.68.201:3100
PUBLIC_FLAG_DISABLE_SIGNUPS=false
```

```yaml
server:
    image: amruthpillai/reactive-resume:server-latest
    restart: always
    ports:
      - 3100:3100
    depends_on:
      - postgres
    environment:
      - PUBLIC_URL=${PUBLIC_URL}
      - PUBLIC_SERVER_URL=${PUBLIC_SERVER_URL}
        [...]
        [..]

client:
    image: amruthpillai/reactive-resume:client-latest
    restart: always
    ports:
      - 3000:3000
    depends_on:
      - server
    environment:
      - PUBLIC_URL=${PUBLIC_URL}
      - PUBLIC_SERVER_URL=${PUBLIC_SERVER_URL}
      - PUBLIC_FLAG_DISABLE_SIGNUPS=${PUBLIC_FLAG_DISABLE_SIGNUPS}
```
```bash
# Trying to register a user
user at 14:01:15 in ~ [I] ➜ curl http://192.168.68.201:3100/auth/register -H "Content-Type: application/json" -d '{"name":"john","email":"john.doe@example.com","username":"john","password":"password"}' -L -v
*   Trying 192.168.68.201:3100...
* Connected to 192.168.68.201 (192.168.68.201) port 3100 (#0)
> POST /auth/register HTTP/1.1
> Host: 192.168.68.201:3100
> User-Agent: curl/7.81.0
> Accept: */*
> Content-Type: application/json
> Content-Length: 99
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 401 Unauthorized
< X-Powered-By: Express
< Vary: Origin
< Access-Control-Allow-Credentials: true
< Content-Type: application/json; charset=utf-8
< Content-Length: 81
< ETag: W/"51-RIZt+OId6cProF0r1sQE2IJcx3I"
< Date: Sat, 26 Aug 2023 21:01:33 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
< 
* Connection #0 to host 192.168.68.201 left intact
{"statusCode":401,"timestamp":"2023-08-26T21:01:33.250Z","path":"/auth/register"}% 
```

I seem to getting a 401 error.

I also tried to login with an existing user, grabbing the JWT and passing it as `Authorization: Bearer <TOKEN>` and I still get the 401 error.

At this point I have exhausted all options.

**Comment by RobertBrown3** [comment]
I have seen similar behavior on multiple browsers. I run reactive resume on Docker and attempt to register. A registration window comes up which I fill in, but when clicking on the "Register" button, nothing happens. It is as if the client is simply not communicating with the server.

This problem has been around for a long time. The last time I ran into it was almost 6 months ago. I came here to report it only to discover that others had already done so. I had hoped that someone would have fixed it by now -- and I am gravely disappointed that nothing has been done about it six months later.  It is a shame that this problem has apparently been ignored for so long.

**Comment by jkrack** [comment]
Same issue - Proxmox > Docker > Portainer > Reactive Resume. 

I can see the UI, the registration window pops up and I enter my information. When I click register, nothing happens. 

I've checked all logs and everything looks good.

**Comment by darsh12** [comment]
Hi,
I'm not sure if this helps but I was facing the same issue after spending hours trying to debug the problem, I discovered it was because that the user was already registered in the database but there was no indication that the user was already registered.

**Comment by AmruthPillai** [comment]
The new version of Reactive Resume is expected to be much much easier to self-host, and I've taken a great deal of care to ensure usability is much much better than before. This includes displaying meaningful error messages from the server when something goes wrong.

It's still in beta and I'm testing it out with a few of my friends, please be patient as I hope to release really soon :)

**Comment by jkrack** [comment]
> Hi, I'm not sure if this helps but I was facing the same issue after spending hours trying to debug the problem, I discovered it was because that the user was already registered in the database but there was no indication that the user was already registered.

So did you just delete the user from the DB? Its unclear on what action you took to remedy the issue.

**Comment by darsh12** [comment]
Yes, I manually deleted the user from the database.

**Comment by chrispe-lab** [comment]
> Same issue - Proxmox > Docker > Portainer > Reactive Resume.
> 
> I can see the UI, the registration window pops up and I enter my information. When I click register, nothing happens.
> 
> I've checked all logs and everything looks good.

I'm getting the same issue.

![image](https://github.com/AmruthPillai/Reactive-Resume/assets/47077770/2abb2b02-4388-4f29-9f76-edb7c8fd9480)

Nothing happens when I click create and no information sent through the logs either. I'm happy to help troubleshoot this.

**Comment by wioch** [comment]
Had the same issue. 
Solved by changing PUBLIC_URL and PUBLIC_SERVER_URL from localhost to host IP address in both server and client sections. 
Example:
- PUBLIC_URL=http://192.168.1.1:3000
- PUBLIC_SERVER_URL=http://192.168.1.1:3100

**Comment by SgtPooki** [comment]
I just tried this on a locally hosted version and i'm getting a 500 error in the browser when I try to register, and the following output from `pnpm dev` in the terminal:

```
[Nest] 80298  - 12/01/2023, 9:47:55 PM   ERROR PrismaClientKnownRequestError:
Invalid `prisma.user.create()` invocation:


The table `public.User` does not exist in the current database.
```

I followed the instructions from `CONTRIBUTING.md` directly.. now looking for some db setup/seed instructions..

**Comment by SgtPooki** [comment]
FYI that I was unblocked by running `pnpm prisma:migrate`. This should be added to CONTRIBUTING.md

**Comment by AmruthPillai** [comment]
This issue was automatically closed as it was created before 2024-01-01.
